### PR TITLE
chore: drop 3.7 and add 3.12

### PR DESCRIPTION
**Pull request by henryiii**
Fixed some warnings when running pre-commit with `-v` too.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -28,11 +28,11 @@ classifiers = [
     "Programming Language :: Python",
     "Programming Language :: Python :: 3",
     "Programming Language :: Python :: 3 :: Only",
-    "Programming Language :: Python :: 3.7",
     "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.8",
```
Comment (eduardo-rodrigues) [inline pyproject.toml:35]:
```suggestion
    "Programming Language :: Python :: 3.11",
```

File: README.rst [hunk 2]
```diff
@@ -81,20 +81,20 @@ physical constants and commonly used constants.
 
 The typical usage is the following:
 
-.. code-block:: python
+.. code-block:: pycon
```
Comment (eduardo-rodrigues) [inline README.rst:84]: ":: pycon"? I'm afraid I never came across that one, funny enough. OK, thanks.

File: README.rst [hunk 3]
```diff
@@ -81,20 +81,20 @@ physical constants and commonly used constants.
...
-    >>> tau_Bs = 1.5 * picosecond    # a particle lifetime, say the Bs meson's
-    >>> ctau_Bs = c_light * tau_Bs   # ctau of the particle, ~450 microns
-    >>> print ctau_Bs                # result in HEP units, so mm
+    >>> from hepunits.units import picosecond, micrometer
+    >>> tau_Bs = 1.5 * picosecond  # a particle lifetime, say the Bs meson's
+    >>> ctau_Bs = c_light * tau_Bs  # ctau of the particle, ~450 microns
+    >>> print(ctau_Bs)  # result in HEP units, so mm
     0.449688687
-    >>> print ctau_Bs / micrometer   # result in micrometers
+    >>> print(ctau_Bs / micrometer)  # result in micrometers
```
Comment (eduardo-rodrigues) [inline README.rst:92]: Wow, how that went unnoticed for so long since moving away from Python 2!? Good catch!

**Review by eduardo-rodrigues — APPROVED** [review]
Many thanks 👍!

File: README.rst (hunk 2 above)
Comment (henryiii) [reply README.rst:84]: pre-commit was swallowing the warning about `>>>` not being valid Python. Switching to pycon (Python Console) caused the warning about the print statement to show up (and the blocks to be formatted).

Comment (eduardo-rodrigues) [reply README.rst:84]: Nice.

**Pull request merged**